internal/logic/logic_test.go

```diff
@@ -120,3 +139,606 @@ func Test_htmlPdfServiceLogic_Ping(t *testing.T) {
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: HtmlToPdf:: GetFile fail",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return(nil, errors.New(""))
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFetchingFile),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: HtmlToPdf:: err unmarshalling json",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return([]byte(""), nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFileParseFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: HtmlToPdf:: failed to decode base64 data",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				js, _ := json.Marshal(map[string]interface{}{
+					"Pages": []interface{}{
+						map[string]interface{}{
+							"Base64PageData": base32.StdEncoding.EncodeToString([]byte("abc")),
+						},
+					},
+				})
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return(js, nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrDecodingData),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: HtmlToPdf:: assertion for Base64PageData failed",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				js, _ := json.Marshal(map[string]interface{}{
+					"Custom-Field": "hello",
+					"Pages": []interface{}{
+						map[string]interface{}{
+							"Base64PageData": map[string]interface{}{},
+						},
+					},
+				})
+				var a uint8
+				a = 10
+				js = append(js, a)
```

Comment:
It was demonstrated to only compare the data that is expected by unmarshaling/decoding to revert any characters that may have been added for formatting and only compare the data.
Change this.

Reply:
done